src/java.base/share/classes/java/lang/reflect/AccessFlag.java

```diff
@@ -187,17 +185,16 @@ public Set<Location> apply(ClassFileFormatVersion cffv) {
      * @apiNote
      * In Java SE 8 and above, the JVM treats the {@code ACC_SUPER}
      * flag as set in every class file (JVMS {@jvms 4.1}).
-     * For class file versions up to Valhalla or if Valhalla is not enabled,
+     * For class file versions up to but not including Valhalla,
      * {@code 0x0020} access flag bit is {@linkplain #SUPER SUPER access flag}; otherwise,
      * the {@code 0x0020} access flag bit is {@linkplain #IDENTITY IDENTITY access flag}.
      */
-    SUPER(0x0000_0020, false,
-            ValhallaFeatures.isEnabled() ? Location.EMPTY_SET : Location.SET_CLASS,
+    SUPER(0x0000_0020, false, Location.EMPTY_SET,
             new Function<ClassFileFormatVersion, Set<Location>>() {
             @Override
             public Set<Location> apply(ClassFileFormatVersion cffv) {
-                return (cffv.compareTo(ClassFileFormatVersion.RELEASE_21) >= 0 &&
-                        ValhallaFeatures.isEnabled()) ? Location.EMPTY_SET : Location.SET_CLASS;}
+                return (cffv.compareTo(ClassFileFormatVersion.RELEASE_22) >= 0)
```

Comment:
Shouldn't it check for preview enabled?

Reply:
The wording of the spec makes it difficult to support SUPER and IDENTITY at the same time flipped on --enable-preview. SUPER is described as being ignored.
It says that 0x0020 means IDENTITY regardless of the class file version number.
The repo hasn't yet sync'd with the mainline changes for jdk23. When that happens, SUPER can be meaningful only for 22 and ignored for 23 and IDENTITY can hold that bit position.